src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java

```diff
@@ -75,6 +88,231 @@ public WorkflowStepFactory(
         stepMap.put(DeleteAgentStep.NAME, () -> new DeleteAgentStep(mlClient));
     }
 
+    /**
+     * Enum encapsulating the different step names, their inputs, outputs, required plugin and timeout of the step
+     */
+
+    public enum WorkflowSteps {
```

Comment:
If you want to leave the stepMap above as-is, there's no reason to have this enum in this class, and it can be put in its own class.  (We could change the map key to the enum value and make it an EnumMap.)

If you want to keep it in this class, we can add a`Supplier<WorkflowStep>` field to each enum value (which would be a good use of the `workflowStep` name you've presently given to a String...) and get rid of the map.

Reply:
This was really a great idea. I removed the stepMap by adding a new field in the enum but just initialized the map in the constructor to avoid breaking anything.